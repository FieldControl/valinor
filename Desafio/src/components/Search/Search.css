.searchMaster {
  width: 100%;
  color: #fff;
  font-size: 25px;
  margin-top: 20px;
  font-weight: 700;
  text-align: center;
}

.search {
  width: 100%;
  display: flex;
  justify-content: center;
}

.search input {
  width: 35%;
  padding: 24px;
  border-radius: 5px;
  border: none;
  background: #e6e6e6;
  font-weight: 700;
}

.search button {
  width: 15%;
  padding: 24px;
  font-weight: 700;
  border-radius: 5px;
  border: none;
  background: #c0c0c0;
  margin-left: 5px;
  cursor: pointer;
  transition: filter 0.2;
}

.search button:hover {
  filter: brightness(0.9);
}
