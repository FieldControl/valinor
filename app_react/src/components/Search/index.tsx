import { useSearch } from "../../hooks/useSearchRepo";

export default function Search() {
  const { setGetRepoName, getRepo, getRepoName } = useSearch()
  return (
    <>
      <input
        type="text"
        id="search_input"
        className="w-52 bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
        placeholder="Buscar"
        required
        onChange={(e) => setGetRepoName(e.target.value)}
      />
      <button
        className="bg-gray-800 hover:bg-gray-500 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
        type="button"
        onClick={() => getRepo(getRepoName)}
      >
        Buscar
      </button>
    </>
  );
}
