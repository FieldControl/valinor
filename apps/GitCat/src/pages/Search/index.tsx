import React, { useEffect, useState } from 'react'
import { useQuery } from '../../contexts/queryContext'

import Header from '../../components/Header'
import NavTypes from '../../components/NavTypes'
import Select from '../../components/Select'
import Card from '../../components/Card'
import NavPages from '../../components/NavPages'

import * as S from './styles'

function Search () {
  const [order, setOrder] = useState({name: 'full_name', display: 'Best Match'})

  const {requestQuerys: {typeSearch, search}, setRequestQuerys, count} = useQuery()
  const selectOptions = [
    {name: 'full_name', display: 'Best Match'},
    {name: 'updated', display: 'Recently Update'}
  ]

  useEffect(() => {
    setRequestQuerys({typeSearch, sort: order.name, page: 1, search})
  }, [order, search, setRequestQuerys, typeSearch])

  return (
    <>
      <Header />
      <S.Content>
        <S.SideMenu>
          <NavTypes />
        </S.SideMenu>
        <S.Main>
          <header>
            <h1>{count} {typeSearch} results</h1>
            <Select value={order} update={setOrder} options={selectOptions} />
          </header>
          <Card />
          <footer>
            <NavPages />
          </footer>
        </S.Main>
      </S.Content>
    </>
  )
}

export default Search