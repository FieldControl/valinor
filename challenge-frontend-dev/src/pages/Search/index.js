import React from 'react';

import SearchList from '../../components/SearchList';

import {
  Container,
  Aside,
  Itens,
  Languages,
  Repositories,
  FooterAside,
} from './styles';

export default function Search() {
  return (
    <Container>
      <Aside>
        <Itens>
          <ul>
            <a href="/">
              <li>
                Repositories<span>1M</span>
              </li>
            </a>
            <hr />
            <a href="/">
              <li>
                Code<span>54M+</span>
              </li>
            </a>
            <hr />
            <a href="/">
              <li>
                Commits<span>5M</span>
              </li>
            </a>
            <hr />
            <a href="/">
              <li>
                Issues<span>1M</span>
              </li>
            </a>
            <hr />
            <a href="/">
              <li>
                Packages<span>2K</span>
              </li>
            </a>
            <hr />
            <a href="/">
              <li>
                Marketplace<span>552</span>
              </li>
            </a>
            <hr />
            <a href="/">
              <li>
                Topics<span>3K</span>
              </li>
            </a>
            <hr />
            <a href="/">
              <li>
                Wikis<span>22K</span>
              </li>
            </a>
            <hr />
            <a href="/">
              <li>
                Users<span>9K</span>
              </li>
            </a>
          </ul>
        </Itens>
        <Languages>
          <h4>Languages</h4>

          <ul>
            <li>
              <a href="/">
                <div>
                  <span>
                    JavaScript<strong>836,321</strong>
                  </span>
                </div>
              </a>
            </li>
            <li>
              <a href="/">
                <div>
                  <span>
                    TypeScript<strong>40,978</strong>
                  </span>
                </div>
              </a>
            </li>
            <li>
              <a href="/">
                <div>
                  <span>
                    HTML<strong>39,126</strong>
                  </span>
                </div>
              </a>
            </li>
            <li>
              <a href="/">
                <div>
                  <span>
                    CSS<strong>26,987</strong>
                  </span>
                </div>
              </a>
            </li>
            <li>
              <a href="/">
                <div>
                  <span>
                    Objective-C<strong>19,638</strong>
                  </span>
                </div>
              </a>
            </li>
            <li>
              <a href="/">
                <div>
                  <span>
                    Java<strong>13,998</strong>
                  </span>
                </div>
              </a>
            </li>
            <li>
              <a href="/">
                <div>
                  <span>
                    Ruby<strong>9,675</strong>
                  </span>
                </div>
              </a>
            </li>
            <li>
              <a href="/">
                <div>
                  <span>
                    Python<strong>5,385</strong>
                  </span>
                </div>
              </a>
            </li>
            <li>
              <a href="/">
                <div>
                  <span>
                    PHP<strong>4,487</strong>
                  </span>
                </div>
              </a>
            </li>
            <li>
              <a href="/">
                <div>
                  <span>
                    C#<strong>3,309</strong>
                  </span>
                </div>
              </a>
            </li>
          </ul>
        </Languages>
        <FooterAside>
          <ul>
            <div>
              <li>
                <a href="/">Advanced search</a>
              </li>
            </div>

            <li>
              <a href="/">Cheat sheet</a>
            </li>
          </ul>
        </FooterAside>
      </Aside>
      <Repositories>
        <SearchList />
      </Repositories>
    </Container>
  );
}
