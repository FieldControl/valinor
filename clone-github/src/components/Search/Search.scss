.search {
  position: relative;
  height: 30px;
  box-sizing: border-box;
  padding: 1px;
  border: 1px solid silver;
  box-shadow: 0 1px 2px rgb(182, 181, 181);
  border-radius: 5px;
  outline: none;
  width: 100%;

  &:focus {
    border-color: rgb(10, 72, 165);
    box-shadow: 0 2px 5px rgb(8, 42, 155);
  }
}
.search__label {
  box-sizing: border-box;
  position: absolute;
  top: 8px;
  left: 5px;
  color: rgb(50, 50, 51);
  z-index: 1;
  opacity: 0.8;
  font-size: 12px;
}
@media (max-width: 820px) {
  .search {
    width: 100%;
  }
}

.search__select {
  width: 100%;
  box-sizing: border-box;
  position: absolute;
  left: 0;
  top: 30px;
  height: 30px;
  color: rgb(255, 255, 255);
  font-size: 1rem;
  background-color: rgb(24, 78, 179);
  letter-spacing: 1px;
  border: 1px solid silver;
}

.search__container {
  position: relative;
  display: flex;
  width: 30%;

  & span {
    position: absolute;
    right: 5px;
    top: 5px;
    font-weight: bold;
    color: blue;
  }
}

@media (max-width: 820px) {
  .search__container {
    width: 60%;
  }
}
