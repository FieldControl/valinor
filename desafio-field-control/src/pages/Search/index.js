import React, { useState, useEffect, useCallback } from 'react';
import {
  FaSearch,
  FaStar,
  FaCodeBranch,
  FaEye,
  FaSpinner,
} from 'react-icons/fa';
import { useLocation, useHistory } from 'react-router-dom';
import {
  Container,
  Input,
  Button,
  InputGroup,
  InputGroupAddon,
  Row,
  Col,
} from 'reactstrap';

import logo from '~/assets/logo.png';
import Pagination from '~/components/Pagination';
import summarizeNumbers from '~/lib/summarizeNumbers';
import api from '~/services/api';

import './styles.css';

function Search() {
  const [repositories, setRepositories] = useState([]);
  const [search, setSearch] = useState('');
  const [page, setPage] = useState(1);
  const [limitPage, setLimitPage] = useState(0);
  const [loading, setLoading] = useState(false);
  const queryParams = useLocation().search;
  const history = useHistory();

  useEffect(() => {
    async function sideEffects() {
      setLoading(true);

      const query = new URLSearchParams(queryParams);
      setSearch(query.get('q'));

      const response = await api('repositories', {
        params: {
          q: query.get('q'),
          page,
          per_page: 10,
        },
      });

      const checkPageLimit = response.data.total_count / 10;

      setLimitPage(checkPageLimit > 100 ? 100 : parseInt(checkPageLimit, 10));
      setRepositories(response.data);
      setLoading(false);
    }

    sideEffects();
  }, [page, queryParams]);

  const handleSearch = useCallback(() => {
    history.replace({
      pathname: '/search',
      search: `?q=${search}`,
    });

    setPage(1);
  }, [history, search]);

  return (
    <>
      <div id="header">
        <a href="/">
          <img className="header-logo" src={logo} alt="Octacat Logo" />
        </a>
        <InputGroup className="mb-2">
          <Input
            id="search"
            name="search"
            placeholder="Pesquisar..."
            onKeyPress={e => e.which === 13 && handleSearch()}
            onChange={event => setSearch(event.target.value)}
            value={search}
            autoFocus
            required
            autoComplete="off"
          />
          <InputGroupAddon addonType="append">
            <Button
              type="submit"
              color="primary"
              onClick={() => handleSearch()}
            >
              <FaSearch />
            </Button>
          </InputGroupAddon>
        </InputGroup>
      </div>
      <Container fluid className="content">
        {!loading ? (
          <Row>
            <Col xs="12" md="12" xl="6">
              {repositories.total_count > 0 ? (
                <>
                  <small className="text-muted">
                    Foi encontrado {summarizeNumbers(repositories.total_count)}{' '}
                    repositório(s)
                  </small>
                  <ul className="repo-list">
                    {repositories.items.map(repository => (
                      <li key={repository.id} className="repo-list-item py-2">
                        <div className="repo-title">
                          <a
                            href={repository.html_url}
                            target="_blank"
                            rel="noopener noreferrer"
                          >
                            {repository.full_name}
                          </a>
                        </div>
                        <p className="mb-1">{repository.description}</p>
                        <div className="info d-flex flex-wrap text-small text-muted text-secondary">
                          <div>
                            <FaStar />{' '}
                            {summarizeNumbers(repository.stargazers_count)}
                          </div>
                          <div>{repository.language}</div>
                          <div>
                            <FaCodeBranch />{' '}
                            {summarizeNumbers(repository.forks_count)}
                          </div>
                          <div>
                            <FaEye /> {summarizeNumbers(repository.watchers)}
                          </div>
                        </div>
                        <hr />
                      </li>
                    ))}
                  </ul>
                </>
              ) : (
                <span>Sua pesquisa não encontrou nenhum repositório</span>
              )}

              <Pagination
                limitPage={limitPage}
                handleClick={setPage}
                currentPage={page}
              />
            </Col>
          </Row>
        ) : (
          <FaSpinner size="36" color="#e07e04" className="loading" />
        )}
      </Container>
    </>
  );
}

export default Search;
