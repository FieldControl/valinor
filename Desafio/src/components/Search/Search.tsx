import { useState } from "react";
import "./Search.css";

const inputSearch = ({ onSearch }) => {
  const [input, setInput] = useState("");

  function handleSearch() {
    const data = {
      input,
    };
    onSearch(data);
  }

  return (
    <div className="searchMaster">
      <form className="search">
        <input
          type="text"
          placeholder="Digite o que deseja pesquisar..."
          value={input}
          onChange={(e) => setInput(e.target.value)}
        />
        <button type="button" onClick={handleSearch}>
          Pesquisar
        </button>
      </form>
    </div>
  );
};

export default inputSearch;
