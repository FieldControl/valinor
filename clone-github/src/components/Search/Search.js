import PropTypes from "prop-types";
import "./Search.scss";
import React, { useState } from "react";

const Search = ({ change, text, changeType, type, click }) => {
  const [select, setSelect] = useState(false);
  return (
    <div className="search__container">
      <label className="search__label" htmlFor="search">
        {text && text.length ? null : "Search or jump to..."}
      </label>
      <input
        id="search"
        data-testid="input"
        onClick={() => setSelect((prevState) => !prevState)}
        className="search"
        onChange={change}
        type="text"
        value={text}
        autoComplete="off"
      />
      {select ? (
        <select
          onChange={changeType}
          className="search__select"
          name="choice"
          value={type}
          onClick={click}
        >
          <option value="issues">issues</option>
          <option value="repositories">repositories</option>
        </select>
      ) : (
        <span data-testid="+">+</span>
      )}
    </div>
  );
};

Search.propTypes = {
  change: PropTypes.func,
  text: PropTypes.string,
  type: PropTypes.string,

  changeType: PropTypes.func,
  click: PropTypes.func,
};

export default Search;
