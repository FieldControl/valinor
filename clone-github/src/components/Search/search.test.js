import { render, fireEvent, cleanup } from "@testing-library/react";
import Search from "./Search";

describe("Search snapshoot", () => {
  afterEach(cleanup);
  test("should take a snapshoot", () => {
    const { asFragment } = render(<Search />);
    expect(asFragment(<Search />)).toMatchSnapshot();
  });
  test("It should change the input", () => {
    const setup = () => {
      const utils = render(<Search />);
      const input = utils.getByTestId("input");
      return {
        input,
        ...utils,
      };
    };
    const { input } = setup();
    fireEvent.change(input, { target: { value: "node" } });
    expect(input.value).toBe("node");
  });

  test("select element  not appear", () => {
    const { getByTestId } = render(<Search />);
    expect(getByTestId("+")).toHaveTextContent("+");
  });
});
