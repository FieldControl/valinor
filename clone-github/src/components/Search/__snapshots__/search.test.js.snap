// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshoot should take a snapshoot 1`] = `
<DocumentFragment>
  <div
    class="search__container"
  >
    <label
      class="search__label"
      for="search"
    >
      Search or jump to...
    </label>
    <input
      autocomplete="off"
      class="search"
      data-testid="input"
      id="search"
      type="text"
      value=""
    />
    <span
      data-testid="+"
    >
      +
    </span>
  </div>
</DocumentFragment>
`;
